From cf3cd327dca39249a536a8372bb331ef0368a6d9 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Thu, 16 Jul 2020 11:08:45 -0500
Subject: [PATCH 46/49] arm64: dts: beacon-imx8mn-kit: Sync Nano with Mini
 Device Trees

There are a bunch changes that evolved over time on the Mini, but
the Nano has been lagging behind.

This patch syncs the Nano board with the Mini.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 .../freescale/beacon-imx8mn-baseboard.dtsi    | 118 ++++--
 .../boot/dts/freescale/beacon-imx8mn-kit.dts  |   3 +-
 .../boot/dts/freescale/beacon-imx8mn-som.dtsi | 383 +++++++++---------
 3 files changed, 279 insertions(+), 225 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mn-baseboard.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mn-baseboard.dtsi
index bebeaba8e752..6d92283699d0 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mn-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mn-baseboard.dtsi
@@ -2,7 +2,41 @@
 /*
  * Copyright 2020 Beacon EmbeddedWorks
  */
+
 / {
+	gpio-keys {
+		compatible = "gpio-keys";
+		autorepeat;
+
+		btn0 {
+			label = "btn0";
+			linux,code = <BTN_0>;
+			gpios = <&pca6416_1 12 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+			wakeup-source;
+		};
+
+		btn1 {
+			label = "btn1";
+			linux,code = <BTN_1>;
+			gpios = <&pca6416_1 13 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+			wakeup-source;
+		};
+
+		btn2 {
+			label = "btn2";
+			linux,code = <BTN_2>;
+			gpios = <&pca6416_1 14 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+			wakeup-source;
+		};
+
+		btn3 {
+			label = "btn3";
+			linux,code = <BTN_3>;
+			gpios = <&pca6416_1 15 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+			wakeup-source;
+		};
+	};
+
 	leds {
 		compatible = "gpio-leds";
 
@@ -33,6 +67,16 @@
 		};
 	};
 
+	reg_camera: regulator-camera {
+		compatible = "regulator-fixed";
+		regulator-name = "mipi_pwr";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&pca6416_1 0 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		startup-delay-us = <100000>;
+	};
+
 	reg_usdhc2_vmmc: regulator-usdhc2 {
 		compatible = "regulator-fixed";
 		regulator-name = "VSD_3V3";
@@ -92,6 +136,10 @@
 	};
 };
 
+&cameradev {
+	status = "okay";
+};
+
 &iomuxc {
 	pinctrl-names = "default";
 
@@ -129,7 +177,7 @@
 				MX8MN_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK		0x82
 				MX8MN_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI		0x82
 				MX8MN_IOMUXC_ECSPI2_MISO_ECSPI2_MISO		0x82
-				MX8MN_IOMUXC_ECSPI1_SS0_GPIO5_IO9		0x41
+				MX8MN_IOMUXC_ECSPI2_SS0_GPIO5_IO13		0x41
 			>;
 		};
 
@@ -163,13 +211,6 @@
 			>;
 		};
 
-		pinctrl_i2c4_gpio: i2c4grp-gpio {
-			fsl,pins = <
-				MX8MN_IOMUXC_I2C4_SCL_GPIO5_IO20        	0x1c3
-				MX8MN_IOMUXC_I2C4_SDA_GPIO5_IO21        	0x1c3
-			>;
-		};
-
 		pinctrl_pcal6414: pcal6414-gpio {
 			fsl,pins = <
 				MX8MN_IOMUXC_SAI2_MCLK_GPIO4_IO27        	0x19
@@ -202,14 +243,16 @@
 
 		pinctrl_uart3: uart3grp {
 			fsl,pins = <
-				MX8MN_IOMUXC_ECSPI1_SCLK_UART3_DCE_RX		0x140
-				MX8MN_IOMUXC_ECSPI1_MOSI_UART3_DCE_TX		0x140
+				MX8MN_IOMUXC_ECSPI1_SCLK_UART3_DCE_RX		0x40
+				MX8MN_IOMUXC_ECSPI1_MOSI_UART3_DCE_TX		0x40
+				MX8MN_IOMUXC_ECSPI1_MISO_UART3_DCE_CTS_B	0x40
+				MX8MN_IOMUXC_ECSPI1_SS0_UART3_DCE_RTS_B	0x40
 			>;
 		};
 
 		pinctrl_usdhc2_gpio: usdhc2grpgpio {
 			fsl,pins = <
-				MX8MN_IOMUXC_SD2_CD_B_USDHC2_CD_B	0x41
+				MX8MN_IOMUXC_SD2_CD_B_GPIO2_IO12	0x41
 				MX8MN_IOMUXC_SD2_RESET_B_GPIO2_IO19	0x41
 			>;
 		};
@@ -251,11 +294,15 @@
 		};
 };
 
+&isi_0 {
+	status = "okay";
+};
+
 &ecspi2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_espi2>;
 	status = "okay";
-	cs-gpios = <&gpio5 9 0>;
+	cs-gpios = <&gpio5 13 GPIO_ACTIVE_LOW>;
 
 	at25@0 {
 		compatible = "atmel,at25";
@@ -271,28 +318,34 @@
 };
 
 &i2c2 {
-	clock-frequency = <400000>;
+	clock-frequency = <384000>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c2>;
 	status = "okay";
 
-	ov5640_mipi: ov5640_mipi@10 {
-		compatible = "ovti,ov5640_mipi";
-		reg = <0x10>;
-		status = "okay";
+	ov5640: ov5640_mipi@3c {
+		compatible = "ovti,ov5640_mipi_v3";
+		reg = <0x3c>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_csi_pwn>, <&pinctrl_csi_rst>;
 		clocks = <&clk IMX8MN_CLK_CLKO1>;
 		clock-names = "csi_mclk";
 		assigned-clocks = <&clk IMX8MN_CLK_CLKO1>;
 		assigned-clock-parents = <&clk IMX8MN_CLK_24M>;
-		assigned-clock-rates = <0>, <24000000>;
+		assigned-clock-rates = <24000000>;
 		csi_id = <0>;
+		DOVDD-supply = <&reg_camera>;
+		AVDD-supply = <&reg_camera>;
+		DVDD-supply = <&reg_camera>;
 		pwn-gpios = <&gpio1 7 GPIO_ACTIVE_HIGH>;
+		rst-gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;
 		mclk = <24000000>;
 		mclk_source = <0>;
+		mipi_csi;
+		status = "okay";
+
 		port {
-			ov5640_mipi1_ep: endpoint {
+			ov5640_ep: endpoint {
 				remote-endpoint = <&mipi1_sensor_ep>;
 			};
 		};
@@ -312,9 +365,11 @@
 	#address-cells = <1>;
 	#size-cells = <0>;
 	status = "okay";
-	port {
-		mipi1_sensor_ep: endpoint1 {
-			remote-endpoint = <&ov5640_mipi1_ep>;
+	mipi-phy-supply = <&ldo3_reg>;
+	port@0 {
+		reg = <0>;
+		mipi1_sensor_ep: endpoint {
+			remote-endpoint = <&ov5640_ep>;
 			data-lanes = <2>;
 			csis-hs-settle = <13>;
 			csis-clk-settle = <2>;
@@ -324,19 +379,17 @@
 };
 
 &i2c4 {
-	clock-frequency = <400000>;
-	pinctrl-names = "default", "gpio";
+	clock-frequency = <384000>;
+	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c4>;
-	pinctrl-1 = <&pinctrl_i2c4_gpio>;
-	scl-gpios = <&gpio5 20 GPIO_ACTIVE_HIGH>;
-	sda-gpios = <&gpio5 21 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 
 	pca6416_0: gpio@20 {
-		compatible = "ti,tca6416";
+		compatible = "nxp,pcal6416";
 		reg = <0x20>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_pcal6414>;
+		vcc-supply = <&buck6_reg>;
 		gpio-controller;
 		#gpio-cells = <2>;
 		interrupt-parent = <&gpio4>;
@@ -344,8 +397,9 @@
 	};
 
 	pca6416_1: gpio@21 {
-		compatible = "ti,tca6416";
+		compatible = "nxp,pcal6416";
 		reg = <0x21>;
+		vcc-supply = <&buck6_reg>;
 		gpio-controller;
 		#gpio-cells = <2>;
 		interrupt-parent = <&gpio4>;
@@ -381,6 +435,11 @@
 	assigned-clock-rates = <393216000>, <361267200>;
 };
 
+&easrc {
+	fsl,asrc-rate  = <48000>;
+	status = "okay";
+};
+
 &sai3 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_sai3>;
@@ -443,6 +502,7 @@
 	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
 	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
 	bus-width = <4>;
+	cd-gpios = <&gpio2 12 GPIO_ACTIVE_LOW>;
 	vmmc-supply = <&reg_usdhc2_vmmc>;
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mn-kit.dts b/arch/arm64/boot/dts/freescale/beacon-imx8mn-kit.dts
index 0a7b5c06cf86..4a89e84c04cc 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mn-kit.dts
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mn-kit.dts
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: (GPL-2.0 OR MIT)
 /*
- * Copyright 2020 Beacon EmbeddedWorks
+ * Copyright 2020 Compass Electronics Group, LLC
  */
 
 /dts-v1/;
@@ -15,7 +15,6 @@
 	compatible = "fsl,imx8mn";
 
 	chosen {
-		bootargs = "console=ttymxc1,115200";
 		stdout-path = &uart2;
 	};
 };
diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
index ec4bc88e0263..ce121964bc74 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mn-som.dtsi
@@ -12,10 +12,6 @@
 		clock-names = "ext_clock";
 		post-power-on-delay-ms = <80>;
 	};
-	
-	memory@40000000 {
-		reg = <0x0 0x40000000 0 0x40000000>;
-	};
 };
 
 &A53_0 {
@@ -27,6 +23,7 @@
 	pinctrl-0 = <&pinctrl_fec1>;
 	phy-mode = "rgmii-id";
 	phy-handle = <&ethphy0>;
+	phy-supply = <&buck6_reg>;
 	fsl,magic-packet;
 	status = "okay";
 
@@ -74,12 +71,8 @@
 		rohm,reset-snvs-powered;
 
 		regulators {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			buck1_reg: regulator@0 {
-				reg = <0>;
-				regulator-compatible = "BUCK1";
+			buck1_reg: BUCK1 {
+				regulator-name = "BUCK1";
 				regulator-min-microvolt = <700000>;
 				regulator-max-microvolt = <1300000>;
 				regulator-boot-on;
@@ -87,9 +80,8 @@
 				regulator-ramp-delay = <1250>;
 			};
 
-			buck2_reg: regulator@1 {
-				reg = <1>;
-				regulator-compatible = "BUCK2";
+			buck2_reg: BUCK2 {
+				regulator-name = "BUCK2";
 				regulator-min-microvolt = <700000>;
 				regulator-max-microvolt = <1300000>;
 				regulator-boot-on;
@@ -99,85 +91,76 @@
 				rohm,dvs-idle-voltage = <900000>;
 			};
 
-			buck3_reg: regulator@2 {
+			buck3_reg: BUCK3 {
 				// BUCK5 in datasheet
-				reg = <2>;
-				regulator-compatible = "BUCK3";
+				regulator-name = "BUCK3";
 				regulator-min-microvolt = <700000>;
 				regulator-max-microvolt = <1350000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck4_reg: regulator@3 {
+			buck4_reg: BUCK4 {
 				// BUCK6 in datasheet
-				reg = <3>;
-				regulator-compatible = "BUCK4";
+				regulator-name = "BUCK4";
 				regulator-min-microvolt = <3000000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck5_reg: regulator@4 {
+			buck5_reg: BUCK5 {
 				// BUCK7 in datasheet
-				reg = <4>;
-				regulator-compatible = "BUCK5";
+				regulator-name = "BUCK5";
 				regulator-min-microvolt = <1605000>;
 				regulator-max-microvolt = <1995000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck6_reg: regulator@5 {
+			buck6_reg: BUCK6 {
 				// BUCK8 in datasheet
-				reg = <5>;
-				regulator-compatible = "BUCK6";
+				regulator-name = "BUCK6";
 				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <1400000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo1_reg: regulator@6 {
-				reg = <6>;
-				regulator-compatible = "LDO1";
-				regulator-min-microvolt = <3000000>;
+			ldo1_reg: LDO1 {
+				regulator-name = "LDO1";
+				regulator-min-microvolt = <1600000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo2_reg: regulator@7 {
-				reg = <7>;
-				regulator-compatible = "LDO2";
-				regulator-min-microvolt = <900000>;
+			ldo2_reg: LDO2 {
+				regulator-name = "LDO2";
+				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <900000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo3_reg: regulator@8 {
-				reg = <8>;
-				regulator-compatible = "LDO3";
+			ldo3_reg: LDO3 {
+				regulator-name = "LDO3";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo4_reg: regulator@9 {
-				reg = <9>;
-				regulator-compatible = "LDO4";
+			ldo4_reg: LDO4 {
+				regulator-name = "LDO4";
 				regulator-min-microvolt = <900000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo6_reg: regulator@11 {
-				reg = <11>;
-				regulator-compatible = "LDO6";
+			ldo6_reg: LDO6 {
+				regulator-name = "LDO6";
 				regulator-min-microvolt = <900000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-boot-on;
@@ -187,6 +170,10 @@
 	};
 };
 
+&gpu {
+	status = "okay";
+};
+
 &i2c3 {
 	clock-frequency = <400000>;
 	pinctrl-names = "default";
@@ -203,6 +190,7 @@
 	rtc@51 {
 		compatible = "nxp,pcf85263";
 		reg = <0x51>;
+		quartz-load-femtofarads = <12500>;
 	};
 };
 
@@ -225,19 +213,26 @@
 	pinctrl-0 = <&pinctrl_uart1>;
 	assigned-clocks = <&clk IMX8MN_CLK_UART1>;
 	assigned-clock-parents = <&clk IMX8MN_SYS_PLL1_80M>;
-	fsl,uart-has-rtscts;
+	uart-has-rtscts;
 	status = "okay";
 
 	bluetooth {
 		compatible = "brcm,bcm43438-bt";
 		shutdown-gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
-		host-wakeup-gpios = <&gpio2 8 GPIO_ACTIVE_HIGH>;
-		device-wakeup-gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
 		clocks = <&osc_32k>;
 		clock-names = "extclk";
+		max-speed = <1000000>;
 	};
 };
 
+&usbmisc1 {
+	vbus-wakeup-supply = <&buck6_reg>;
+};
+
+&usbphynop1 {
+	vcc-supply = <&buck6_reg>;
+};
+
 &usdhc1 {
 	#address-cells = <1>;
 	#size-cells = <0>;
@@ -249,6 +244,7 @@
 	pm-ignore-notify;
 	keep-power-in-suspend;
 	mmc-pwrseq = <&usdhc1_pwrseq>;
+	vqmmc-supply = <&buck5_reg>;
 	status = "okay";
 
 	brcmf: bcrmf@1 {
@@ -280,168 +276,167 @@
 };
 
 &iomuxc {
+	pinctrl_fec1: fec1grp {
+		fsl,pins = <
+			MX8MN_IOMUXC_ENET_MDC_ENET1_MDC		0x3
+			MX8MN_IOMUXC_ENET_MDIO_ENET1_MDIO	0x3
+			MX8MN_IOMUXC_ENET_TD3_ENET1_RGMII_TD3	0x1f
+			MX8MN_IOMUXC_ENET_TD2_ENET1_RGMII_TD2	0x1f
+			MX8MN_IOMUXC_ENET_TD1_ENET1_RGMII_TD1	0x1f
+			MX8MN_IOMUXC_ENET_TD0_ENET1_RGMII_TD0	0x1f
+			MX8MN_IOMUXC_ENET_RD3_ENET1_RGMII_RD3	0x91
+			MX8MN_IOMUXC_ENET_RD2_ENET1_RGMII_RD2	0x91
+			MX8MN_IOMUXC_ENET_RD1_ENET1_RGMII_RD1	0x91
+			MX8MN_IOMUXC_ENET_RD0_ENET1_RGMII_RD0	0x91
+			MX8MN_IOMUXC_ENET_TXC_ENET1_RGMII_TXC	0x1f
+			MX8MN_IOMUXC_ENET_RXC_ENET1_RGMII_RXC	0x91
+			MX8MN_IOMUXC_ENET_RX_CTL_ENET1_RGMII_RX_CTL	0x91
+			MX8MN_IOMUXC_ENET_TX_CTL_ENET1_RGMII_TX_CTL	0x1f
+			MX8MN_IOMUXC_SAI2_RXC_GPIO4_IO22	0x19
+		>;
+	};
 
-		pinctrl_fec1: fec1grp {
-			fsl,pins = <
-				MX8MN_IOMUXC_ENET_MDC_ENET1_MDC		0x3
-				MX8MN_IOMUXC_ENET_MDIO_ENET1_MDIO	0x3
-				MX8MN_IOMUXC_ENET_TD3_ENET1_RGMII_TD3	0x1f
-				MX8MN_IOMUXC_ENET_TD2_ENET1_RGMII_TD2	0x1f
-				MX8MN_IOMUXC_ENET_TD1_ENET1_RGMII_TD1	0x1f
-				MX8MN_IOMUXC_ENET_TD0_ENET1_RGMII_TD0	0x1f
-				MX8MN_IOMUXC_ENET_RD3_ENET1_RGMII_RD3	0x91
-				MX8MN_IOMUXC_ENET_RD2_ENET1_RGMII_RD2	0x91
-				MX8MN_IOMUXC_ENET_RD1_ENET1_RGMII_RD1	0x91
-				MX8MN_IOMUXC_ENET_RD0_ENET1_RGMII_RD0	0x91
-				MX8MN_IOMUXC_ENET_TXC_ENET1_RGMII_TXC	0x1f
-				MX8MN_IOMUXC_ENET_RXC_ENET1_RGMII_RXC	0x91
-				MX8MN_IOMUXC_ENET_RX_CTL_ENET1_RGMII_RX_CTL	0x91
-				MX8MN_IOMUXC_ENET_TX_CTL_ENET1_RGMII_TX_CTL	0x1f
-				MX8MN_IOMUXC_SAI2_RXC_GPIO4_IO22	0x19
-			>;
-		};
-
-		pinctrl_flexspi0: flexspi0grp {
-			fsl,pins = <
-				MX8MN_IOMUXC_NAND_ALE_QSPI_A_SCLK		0x1c2
-				MX8MN_IOMUXC_NAND_CE0_B_QSPI_A_SS0_B		0x82
-				MX8MN_IOMUXC_NAND_DATA00_QSPI_A_DATA0		0x82
-				MX8MN_IOMUXC_NAND_DATA01_QSPI_A_DATA1		0x82
-				MX8MN_IOMUXC_NAND_DATA02_QSPI_A_DATA2		0x82
-				MX8MN_IOMUXC_NAND_DATA03_QSPI_A_DATA3		0x82
-			>;
-		};
+	pinctrl_flexspi0: flexspi0grp {
+		fsl,pins = <
+			MX8MN_IOMUXC_NAND_ALE_QSPI_A_SCLK		0x1c2
+			MX8MN_IOMUXC_NAND_CE0_B_QSPI_A_SS0_B		0x82
+			MX8MN_IOMUXC_NAND_DATA00_QSPI_A_DATA0		0x82
+			MX8MN_IOMUXC_NAND_DATA01_QSPI_A_DATA1		0x82
+			MX8MN_IOMUXC_NAND_DATA02_QSPI_A_DATA2		0x82
+			MX8MN_IOMUXC_NAND_DATA03_QSPI_A_DATA3		0x82
+		>;
+	};
 
-		pinctrl_i2c1: i2c1grp {
-			fsl,pins = <
-				MX8MN_IOMUXC_I2C1_SCL_I2C1_SCL			0x400001c3
-				MX8MN_IOMUXC_I2C1_SDA_I2C1_SDA			0x400001c3
-			>;
-		};
+	pinctrl_i2c1: i2c1grp {
+		fsl,pins = <
+			MX8MN_IOMUXC_I2C1_SCL_I2C1_SCL			0x400001c3
+			MX8MN_IOMUXC_I2C1_SDA_I2C1_SDA			0x400001c3
+		>;
+	};
 
-		pinctrl_i2c3: i2c3grp {
-			fsl,pins = <
-				MX8MN_IOMUXC_I2C3_SCL_I2C3_SCL			0x400001c3
-				MX8MN_IOMUXC_I2C3_SDA_I2C3_SDA			0x400001c3
-			>;
-		};
+	pinctrl_i2c3: i2c3grp {
+		fsl,pins = <
+			MX8MN_IOMUXC_I2C3_SCL_I2C3_SCL			0x400001c3
+			MX8MN_IOMUXC_I2C3_SDA_I2C3_SDA			0x400001c3
+		>;
+	};
 
-		pinctrl_pmic: pmicirq {
-			fsl,pins = <
-				MX8MN_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
-			>;
-		};
+	pinctrl_pmic: pmicirq {
+		fsl,pins = <
+			MX8MN_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
+		>;
+	};
 
-		pinctrl_uart1: uart1grp {
-			fsl,pins = <
-				MX8MN_IOMUXC_UART1_RXD_UART1_DCE_RX	0x140
-				MX8MN_IOMUXC_UART1_TXD_UART1_DCE_TX	0x140
-				MX8MN_IOMUXC_UART3_RXD_UART1_DCE_CTS_B	0x140
-				MX8MN_IOMUXC_UART3_TXD_UART1_DCE_RTS_B	0x140
-				MX8MN_IOMUXC_SD1_DATA4_GPIO2_IO6	0x19
-				MX8MN_IOMUXC_SD1_DATA5_GPIO2_IO7	0x19
-				MX8MN_IOMUXC_SD1_DATA6_GPIO2_IO8	0x19
-				MX8MN_IOMUXC_GPIO1_IO00_ANAMIX_REF_CLK_32K	0x141
-			>;
-		};
+	pinctrl_uart1: uart1grp {
+		fsl,pins = <
+			MX8MN_IOMUXC_UART1_RXD_UART1_DCE_RX	0x140
+			MX8MN_IOMUXC_UART1_TXD_UART1_DCE_TX	0x140
+			MX8MN_IOMUXC_UART3_RXD_UART1_DCE_CTS_B	0x140
+			MX8MN_IOMUXC_UART3_TXD_UART1_DCE_RTS_B	0x140
+			MX8MN_IOMUXC_SD1_DATA4_GPIO2_IO6	0x100
+			MX8MN_IOMUXC_SD1_DATA5_GPIO2_IO7	0x140
+			MX8MN_IOMUXC_SD1_DATA6_GPIO2_IO8	0x140
+			MX8MN_IOMUXC_GPIO1_IO00_ANAMIX_REF_CLK_32K	0x141
+		>;
+	};
 
-		pinctrl_usdhc1_gpio: usdhc1grpgpio {
-			fsl,pins = <
-				MX8MN_IOMUXC_SD1_RESET_B_GPIO2_IO10	0x41
-			>;
-		};
+	pinctrl_usdhc1_gpio: usdhc1grpgpio {
+		fsl,pins = <
+			MX8MN_IOMUXC_SD1_RESET_B_GPIO2_IO10	0x41
+		>;
+	};
 
-		pinctrl_usdhc1: usdhc1grp {
-			fsl,pins = <
-				MX8MN_IOMUXC_SD1_CLK_USDHC1_CLK		0x190
-				MX8MN_IOMUXC_SD1_CMD_USDHC1_CMD		0x1d0
-				MX8MN_IOMUXC_SD1_DATA0_USDHC1_DATA0	0x1d0
-				MX8MN_IOMUXC_SD1_DATA1_USDHC1_DATA1	0x1d0
-				MX8MN_IOMUXC_SD1_DATA2_USDHC1_DATA2	0x1d0
-				MX8MN_IOMUXC_SD1_DATA3_USDHC1_DATA3	0x1d0
-			>;
-		};
+	pinctrl_usdhc1: usdhc1grp {
+		fsl,pins = <
+			MX8MN_IOMUXC_SD1_CLK_USDHC1_CLK		0x190
+			MX8MN_IOMUXC_SD1_CMD_USDHC1_CMD		0x1d0
+			MX8MN_IOMUXC_SD1_DATA0_USDHC1_DATA0	0x1d0
+			MX8MN_IOMUXC_SD1_DATA1_USDHC1_DATA1	0x1d0
+			MX8MN_IOMUXC_SD1_DATA2_USDHC1_DATA2	0x1d0
+			MX8MN_IOMUXC_SD1_DATA3_USDHC1_DATA3	0x1d0
+		>;
+	};
 
-		pinctrl_usdhc1_100mhz: usdhc1grp100mhz {
-			fsl,pins = <
-				MX8MN_IOMUXC_SD1_CLK_USDHC1_CLK		0x194
-				MX8MN_IOMUXC_SD1_CMD_USDHC1_CMD		0x1d4
-				MX8MN_IOMUXC_SD1_DATA0_USDHC1_DATA0	0x1d4
-				MX8MN_IOMUXC_SD1_DATA1_USDHC1_DATA1	0x1d4
-				MX8MN_IOMUXC_SD1_DATA2_USDHC1_DATA2	0x1d4
-				MX8MN_IOMUXC_SD1_DATA3_USDHC1_DATA3	0x1d4
-			>;
-		};
+	pinctrl_usdhc1_100mhz: usdhc1grp100mhz {
+		fsl,pins = <
+			MX8MN_IOMUXC_SD1_CLK_USDHC1_CLK		0x194
+			MX8MN_IOMUXC_SD1_CMD_USDHC1_CMD		0x1d4
+			MX8MN_IOMUXC_SD1_DATA0_USDHC1_DATA0	0x1d4
+			MX8MN_IOMUXC_SD1_DATA1_USDHC1_DATA1	0x1d4
+			MX8MN_IOMUXC_SD1_DATA2_USDHC1_DATA2	0x1d4
+			MX8MN_IOMUXC_SD1_DATA3_USDHC1_DATA3	0x1d4
+		>;
+	};
 
-		pinctrl_usdhc1_200mhz: usdhc1grp200mhz {
-			fsl,pins = <
-				MX8MN_IOMUXC_SD1_CLK_USDHC1_CLK		0x196
-				MX8MN_IOMUXC_SD1_CMD_USDHC1_CMD		0x1d6
-				MX8MN_IOMUXC_SD1_DATA0_USDHC1_DATA0	0x1d6
-				MX8MN_IOMUXC_SD1_DATA1_USDHC1_DATA1	0x1d6
-				MX8MN_IOMUXC_SD1_DATA2_USDHC1_DATA2	0x1d6
-				MX8MN_IOMUXC_SD1_DATA3_USDHC1_DATA3	0x1d6
-			>;
-		};
+	pinctrl_usdhc1_200mhz: usdhc1grp200mhz {
+		fsl,pins = <
+			MX8MN_IOMUXC_SD1_CLK_USDHC1_CLK		0x196
+			MX8MN_IOMUXC_SD1_CMD_USDHC1_CMD		0x1d6
+			MX8MN_IOMUXC_SD1_DATA0_USDHC1_DATA0	0x1d6
+			MX8MN_IOMUXC_SD1_DATA1_USDHC1_DATA1	0x1d6
+			MX8MN_IOMUXC_SD1_DATA2_USDHC1_DATA2	0x1d6
+			MX8MN_IOMUXC_SD1_DATA3_USDHC1_DATA3	0x1d6
+		>;
+	};
 
-		pinctrl_usdhc3: usdhc3grp {
-			fsl,pins = <
-				MX8MN_IOMUXC_NAND_WE_B_USDHC3_CLK		0x190
-				MX8MN_IOMUXC_NAND_WP_B_USDHC3_CMD		0x1d0
-				MX8MN_IOMUXC_NAND_DATA04_USDHC3_DATA0		0x1d0
-				MX8MN_IOMUXC_NAND_DATA05_USDHC3_DATA1		0x1d0
-				MX8MN_IOMUXC_NAND_DATA06_USDHC3_DATA2		0x1d0
-				MX8MN_IOMUXC_NAND_DATA07_USDHC3_DATA3		0x1d0
-				MX8MN_IOMUXC_NAND_RE_B_USDHC3_DATA4		0x1d0
-				MX8MN_IOMUXC_NAND_CE2_B_USDHC3_DATA5		0x1d0
-				MX8MN_IOMUXC_NAND_CE3_B_USDHC3_DATA6		0x1d0
-				MX8MN_IOMUXC_NAND_CLE_USDHC3_DATA7		0x1d0
-				MX8MN_IOMUXC_NAND_CE1_B_USDHC3_STROBE 		0x190
-			>;
-		};
+	pinctrl_usdhc3: usdhc3grp {
+		fsl,pins = <
+			MX8MN_IOMUXC_NAND_WE_B_USDHC3_CLK		0x190
+			MX8MN_IOMUXC_NAND_WP_B_USDHC3_CMD		0x1d0
+			MX8MN_IOMUXC_NAND_DATA04_USDHC3_DATA0		0x1d0
+			MX8MN_IOMUXC_NAND_DATA05_USDHC3_DATA1		0x1d0
+			MX8MN_IOMUXC_NAND_DATA06_USDHC3_DATA2		0x1d0
+			MX8MN_IOMUXC_NAND_DATA07_USDHC3_DATA3		0x1d0
+			MX8MN_IOMUXC_NAND_RE_B_USDHC3_DATA4		0x1d0
+			MX8MN_IOMUXC_NAND_CE2_B_USDHC3_DATA5		0x1d0
+			MX8MN_IOMUXC_NAND_CE3_B_USDHC3_DATA6		0x1d0
+			MX8MN_IOMUXC_NAND_CLE_USDHC3_DATA7		0x1d0
+			MX8MN_IOMUXC_NAND_CE1_B_USDHC3_STROBE		0x190
+		>;
+	};
 
-		pinctrl_usdhc3_100mhz: usdhc3grp100mhz {
-			fsl,pins = <
-				MX8MN_IOMUXC_NAND_WE_B_USDHC3_CLK		0x194
-				MX8MN_IOMUXC_NAND_WP_B_USDHC3_CMD		0x1d4
-				MX8MN_IOMUXC_NAND_DATA04_USDHC3_DATA0		0x1d4
-				MX8MN_IOMUXC_NAND_DATA05_USDHC3_DATA1		0x1d4
-				MX8MN_IOMUXC_NAND_DATA06_USDHC3_DATA2		0x1d4
-				MX8MN_IOMUXC_NAND_DATA07_USDHC3_DATA3		0x1d4
-				MX8MN_IOMUXC_NAND_RE_B_USDHC3_DATA4		0x1d4
-				MX8MN_IOMUXC_NAND_CE2_B_USDHC3_DATA5		0x1d4
-				MX8MN_IOMUXC_NAND_CE3_B_USDHC3_DATA6		0x1d4
-				MX8MN_IOMUXC_NAND_CLE_USDHC3_DATA7		0x1d4
-				MX8MN_IOMUXC_NAND_CE1_B_USDHC3_STROBE 		0x194
-			>;
-		};
+	pinctrl_usdhc3_100mhz: usdhc3grp100mhz {
+		fsl,pins = <
+			MX8MN_IOMUXC_NAND_WE_B_USDHC3_CLK		0x194
+			MX8MN_IOMUXC_NAND_WP_B_USDHC3_CMD		0x1d4
+			MX8MN_IOMUXC_NAND_DATA04_USDHC3_DATA0		0x1d4
+			MX8MN_IOMUXC_NAND_DATA05_USDHC3_DATA1		0x1d4
+			MX8MN_IOMUXC_NAND_DATA06_USDHC3_DATA2		0x1d4
+			MX8MN_IOMUXC_NAND_DATA07_USDHC3_DATA3		0x1d4
+			MX8MN_IOMUXC_NAND_RE_B_USDHC3_DATA4		0x1d4
+			MX8MN_IOMUXC_NAND_CE2_B_USDHC3_DATA5		0x1d4
+			MX8MN_IOMUXC_NAND_CE3_B_USDHC3_DATA6		0x1d4
+			MX8MN_IOMUXC_NAND_CLE_USDHC3_DATA7		0x1d4
+			MX8MN_IOMUXC_NAND_CE1_B_USDHC3_STROBE		0x194
+		>;
+	};
 
-		pinctrl_usdhc3_200mhz: usdhc3grp200mhz {
-			fsl,pins = <
-				MX8MN_IOMUXC_NAND_WE_B_USDHC3_CLK		0x196
-				MX8MN_IOMUXC_NAND_WP_B_USDHC3_CMD		0x1d6
-				MX8MN_IOMUXC_NAND_DATA04_USDHC3_DATA0		0x1d6
-				MX8MN_IOMUXC_NAND_DATA05_USDHC3_DATA1		0x1d6
-				MX8MN_IOMUXC_NAND_DATA06_USDHC3_DATA2		0x1d6
-				MX8MN_IOMUXC_NAND_DATA07_USDHC3_DATA3		0x1d6
-				MX8MN_IOMUXC_NAND_RE_B_USDHC3_DATA4		0x1d6
-				MX8MN_IOMUXC_NAND_CE2_B_USDHC3_DATA5		0x1d6
-				MX8MN_IOMUXC_NAND_CE3_B_USDHC3_DATA6		0x1d6
-				MX8MN_IOMUXC_NAND_CLE_USDHC3_DATA7		0x1d6
-				MX8MN_IOMUXC_NAND_CE1_B_USDHC3_STROBE 		0x196
-			>;
-		};
+	pinctrl_usdhc3_200mhz: usdhc3grp200mhz {
+		fsl,pins = <
+			MX8MN_IOMUXC_NAND_WE_B_USDHC3_CLK		0x196
+			MX8MN_IOMUXC_NAND_WP_B_USDHC3_CMD		0x1d6
+			MX8MN_IOMUXC_NAND_DATA04_USDHC3_DATA0		0x1d6
+			MX8MN_IOMUXC_NAND_DATA05_USDHC3_DATA1		0x1d6
+			MX8MN_IOMUXC_NAND_DATA06_USDHC3_DATA2		0x1d6
+			MX8MN_IOMUXC_NAND_DATA07_USDHC3_DATA3		0x1d6
+			MX8MN_IOMUXC_NAND_RE_B_USDHC3_DATA4		0x1d6
+			MX8MN_IOMUXC_NAND_CE2_B_USDHC3_DATA5		0x1d6
+			MX8MN_IOMUXC_NAND_CE3_B_USDHC3_DATA6		0x1d6
+			MX8MN_IOMUXC_NAND_CLE_USDHC3_DATA7		0x1d6
+			MX8MN_IOMUXC_NAND_CE1_B_USDHC3_STROBE		0x196
+		>;
+	};
 
-		pinctrl_wdog: wdoggrp {
-			fsl,pins = <
-				MX8MN_IOMUXC_GPIO1_IO02_WDOG1_WDOG_B		0xc6
-			>;
-		};
+	pinctrl_wdog: wdoggrp {
+		fsl,pins = <
+			MX8MN_IOMUXC_GPIO1_IO02_WDOG1_WDOG_B		0xc6
+		>;
+	};
 
-		pinctrl_wlan: wlangrp {
-			fsl,pins = <
-				MX8MN_IOMUXC_SD1_DATA7_GPIO2_IO9		0x111
-			>;
-		};
+	pinctrl_wlan: wlangrp {
+		fsl,pins = <
+			MX8MN_IOMUXC_SD1_DATA7_GPIO2_IO9		0x111
+		>;
+	};
 };
 
-- 
2.17.1

